proof/crefine/ARM_HYP/IsolatedThreadAction.thy

```diff
@@ -212,36 +213,49 @@ lemma isolate_thread_actions_asUser:
   apply (clarsimp simp: partial_overwrite_get_tcb_state_regs
                         put_tcb_state_regs_ko_at')
   apply (case_tac ko, simp)
+  apply (rename_tac uc)
+  apply (case_tac uc, simp add: modify_registers_def atcbContextGet_def atcbContextSet_def)
   done
 
-context begin interpretation Arch . (*FIXME: arch_split*)
-
 lemma getRegister_simple:
-  "getRegister r = (\<lambda>con. ({(con r, con)}, False))"
+  "getRegister r = (\<lambda>con. ({(user_regs con r, con)}, False))"
```

Comment:
Had forgotten that we did these monad unfoldings here. Those are going to bite when we move to the concurrency monad, but that's a problem for future us. Nothing to do with this PR in any case.

Reply:
I don't know anything about how they're used but I don't think that it will be hard to write a similar unfolding for the trace monad. It would be a bit fiddly if we couldn't make the function atomic but even that shouldn't be too bad.